File: python/report.py
report = [
    {'enrollment': 'rit2011001', 'name': 'Julia', 'subject': [
        {'code': 'NLP', 'grade': 'A'}, {'code': 'DSA', 'grade': 'A'}]},
    {'enrollment': 'rit2011001', 'name': 'Samantha', 'subject': [
        {'code': 'COM', 'grade': 'B'}, {'code': 'OOM', 'grade': 'A'}, {'code': 'DSA', 'grade': 'A'}]}
]
inside = {}
new_report = []
for i in range(len(report)):
    # print(report[i]['subject'])
    for j in range(len(report[i]['subject'])):
        inside['code'] = report[i]['subject'][j]['code']
        inside['grade'] = report[i]['subject'][j]['grade']
        inside['enrollment'] = report[i]['enrollment']
        inside['name'] = report[i]['name']
        new_report.append(inside)
        inside = {}

reports = sorted(new_report, key=lambda report: report['code'])
for i in range(len(reports)):
    for k, v in reports[i].items():
        print(v, end=' ')
    print()

File: cpp/report.swift
//Load the file into an NSData object called JSONData

NSError *error = nil;

NSData *JSONData = [NSData dataWithContentsOfFile:filePath options:NSDataReadingMappedIfSafe error:&error];

// Create an Objective-C object from JSON Data

id JSONObject = [NSJSONSerialization
    JSONObjectWithData:JSONData
               options:NSJSONReadingAllowFragments
                 error:&error];
// NSLog(@"%@", JSONObject);
NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
NSMutableArray *newReport = [[NSMutableArray alloc] init];
NSDictionary *report = JSONObject[@"report"];
for (id myArrayElement in report)
{
    NSDictionary *subject = myArrayElement[@"subject"];
    for (id mySubjectElement in subject)
    {
        dict[@"code"] = mySubjectElement[@"code"];
        dict[@"grade"] = mySubjectElement[@"grade"];
        dict[@"enrollment"] = myArrayElement[@"enrollment"];
        dict[@"name"] = myArrayElement[@"name"];
        [newReport addObject:dict];
        // dict = [String: String]();
    }
};
// NSLog(@"%@", newReport);
NSSortDescriptor *sd = [[NSSortDescriptor alloc] initWithKey:@"code" ascending:YES];
[newReport sortedArrayUsingDescriptors:@[ sd ]];
for (id report in newReport)
{
    NSEnumerator *enu = [report keyEnumerator];
    for (NSString *key in enu)
    {
        // NSLog(@"key : %@",key);
        NSLog(@"%@", [[report valueForKey:key] string]);
    }
};